frontend/src/components/mainPage/codeEditor/subcomponents/Codearea.vue

```diff
@@ -42,6 +43,72 @@ export default {
             this.$emit("pickVariableEvent", variable);
         },
 
+        moveTrackedVariables(changes) {
+            const handleVarTrackerMove = (ch, varObj) => {
+                if (ch.size > 0) {
+                    if (ch.start > varObj.start && ch.start < varObj.end) return "Delete";
+                    if (ch.start <= varObj.start) {
+                        varObj.start += ch.size;
+                        varObj.end += ch.size;
+                    }
+                } else {
+                    if (ch.start >= varObj.start && ch.start < varObj.end) return "Delete";
+                    if (ch.end > varObj.start && ch.end < varObj.end) return "Delete";
+                    if (ch.start <= varObj.start) {
+                        varObj.start += ch.size;
+                        varObj.end += ch.size;
+                    }
+                }
+            };
+
```

Comment:
Powinniśmy unikać modyfikowania obiektu ze store (this.project) bezpośrednio w komponentach. Wszelkie zmieny tego obiektu powinny być realizowane przez actions, które wywołują mutations w pliku project.js. (Polecam zapoznanie się z VueX).